test/integration/Cover/expireCover.js

```diff
@@ -0,0 +1,170 @@
+const { expect } = require('chai');
+const { ethers } = require('hardhat');
+
+const { AddressZero, MaxUint256 } = ethers.constants;
+const { increaseTime } = require('../utils').evm;
+const { daysToSeconds } = require('../utils').helpers;
+const { calculateFirstTrancheId } = require('../utils/staking');
+const { BigNumber } = require('ethers');
+
+const { parseEther } = ethers.utils;
+
+const allocationAmount = BigNumber.from(472);
+
+const stakedProductParamTemplate = {
+  productId: 1,
+  recalculateEffectiveWeight: true,
+  setTargetWeight: true,
+  targetWeight: 100,
+  setTargetPrice: true,
+  targetPrice: 100,
+};
+
+function sum(arr) {
+  return arr.reduce((x, y) => x.add(y), BigNumber.from(0));
+}
+
+describe('expireCover', function () {
+  const buyCoverFixture = {
+    coverId: 0,
+    owner: AddressZero,
+    productId: 1,
+    coverAsset: 0,
+    amount: parseEther('1'),
+    period: daysToSeconds((Date.now() / 1000 / 60 / 60 / 24) % 28 ? 28 : 29),
+    maxPremiumInAsset: MaxUint256,
+    paymentAsset: 0,
+    commissionRatio: 0,
+    commissionDestination: AddressZero,
+    ipfsData: 'ipfs data',
+  };
+
+  beforeEach(async function () {
+    const { tk: nxm, stakingProducts, stakingPool1, stakingPool2, stakingPool3, tc: tokenController } = this.contracts;
+    const staker = this.accounts.defaultSender;
+    const [manager1, manager2, manager3] = this.accounts.stakingPoolManagers;
+    const stakeAmount = parseEther('9000000');
+
+    await stakingProducts.connect(manager1).setProducts(1, [stakedProductParamTemplate]);
+    await stakingProducts.connect(manager2).setProducts(2, [stakedProductParamTemplate]);
+    await stakingProducts.connect(manager3).setProducts(3, [stakedProductParamTemplate]);
+
+    // stake
+    const firstActiveTrancheId = await calculateFirstTrancheId(
+      await ethers.provider.getBlock('latest'),
+      buyCoverFixture.period,
+      0,
+    );
+
+    await nxm.approve(tokenController.address, MaxUint256);
+    await stakingPool1.connect(staker).depositTo(stakeAmount, firstActiveTrancheId + 5, 0, staker.address);
+    await stakingPool2.connect(staker).depositTo(stakeAmount, firstActiveTrancheId + 5, 0, staker.address);
+    await stakingPool3.connect(staker).depositTo(stakeAmount, firstActiveTrancheId + 5, 0, staker.address);
+  });
+
+  it('should revert when cover is not expired', async function () {
+    const { cover } = this.contracts;
+    const [coverBuyer] = this.accounts.members;
+    const { amount } = buyCoverFixture;
+    const coverBuyerAddress = await coverBuyer.getAddress();
+
+    await cover
+      .connect(coverBuyer)
+      .buyCover(
+        { ...buyCoverFixture, commissionDestination: coverBuyerAddress, owner: coverBuyerAddress },
```

Comment:
Why do you need to specify the commission destination and owner and not use the default from the template?